flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointFailureReason.java

```diff
@@ -79,4 +79,38 @@
 	public String message() {
 		return message;
 	}
+
+	/**
+	 * @return true if this value indicates a failure reason happening before a checkpoint is passed to a job's tasks.
+	 */
+	public boolean isPreFlight() {
```

Comment:
@1u0 Although better than before, I think that this method is also fragile as it expects that anybody who adds a new member here, will have to update the method accordingly. As I said in the previous comment, I find the solution of a `final boolean` flag which is set in the constructor more future-proof.

Reply:
Imo, it is the same in regard of being future-proof. Currently I'm checking at runtime (by a test) that this values are defined for all enum's values.

If someone adds a new enum value, they would need to think and define if it's `isPreFlight` or not in both approaches.

I personally find the approach with separate method less intrusive and more friendlier when you need to delete such attribute. It's also more friendlier (densely located), when you want to see the attribute's values by enum's value.

Please let me know if this is a blocker from your point of view, I'd change it to constructor/field based approach.

Reply:
I still believe that the flag is a better approach. Imagine if in the future we want to add another attribute. Then we will have to add another method that enumerates all the exceptions and another test that checks the completeness of the method.